nmmo/systems/ai/utils.py

```diff
@@ -8,18 +8,15 @@
 
 from nmmo.lib.utils import in_bounds
 
-
 def validTarget(ent, targ, rng):
-   if targ is None or not targ.alive:
-      return False
-   if lInfty(ent.pos, targ.pos) > rng:
+   if targ is None or not targ.alive or lInfty(ent.pos, targ.pos) > rng:
       return False
    return True
 
 
 def validResource(ent, tile, rng):
    return tile is not None and tile.state.tex in (
-      'foilage', 'water') and distance(ent, tile) <= rng
+      'foilage', 'water') and lInfty(ent.pos, tile.pos) <= rng
```

Comment:
can we use `utils.linf_single` here? 

there are several distance functions defined, so it'd be great if we can remove redundant ones